File: app/src/main/java/nu/parley/ui/SplashActivity.java
package nu.parley.ui;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import nu.parley.BuildConfig;
import nu.parley.R;

public final class SplashActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        openIdentifierActivity();
    }

    private void openIdentifierActivity() {
        new Handler().postDelayed(() -> {
            Intent intent = new Intent(SplashActivity.this, IdentifierActivity.class);

            startActivity(intent);
        }, BuildConfig.DEBUG ? 1000 : 2500);
    }
}

File: app/src/main/java/nu/parley/ui/SplashActivity.kt
package nu.parley.ui

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import nu.parley.BuildConfig
import nu.parley.R
import nu.parley.util.BaseActivity

@SuppressLint("CustomSplashScreen")
class SplashActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        openIdentifierActivity()
    }

    private fun openIdentifierActivity() {
        Handler().postDelayed({
            val intent =
                Intent(this@SplashActivity, IdentifierActivity::class.java)
            startActivity(intent)
        }, if (BuildConfig.DEBUG) 1L else 2500L)
    }
}